### ISSUE TITLE
Provide a way to replace routes

### ISSUE THREAD

**Issue by Zazo032** [csells/go_router#60]
Currently, there's no `.pushReplacement` compatible API for replacing the current route

**Comment by Moussenger** [comment]
You can replace the current route just routing to a new one.

If you are in `/path/1/other` and wanted to replace `other` by `new`, you can just navigate to `/path/1/new`. Also, if you want to replace `1/other' by just `2`, you can just navigate to `/path/2`.

Is this what you are looking for?

**Comment by Moussenger** [comment]
Yes. You are going to generate a new stack with each new route that doesn't match with the previous one.

If you are in `/` and wanted to go `/login`, you will go to that page being replaced the route. If you want to have the back stack, then you need to declarete `/login` as a subroute of `/`. This way, when you go to `/login`, you will generate an stack with `/` and then `/login`.

You can check the [named routes example](https://github.com/csells/go_router/blob/master/example/lib/named_routes.dart)  to see how this works with subroutes generating stack and a login page that replace it.

**Comment by Zazo032** [comment]
Thanks! Will try that 😁
